### DISCUSSION TITLE
How to request a fixture in another fixture, without passing it as a parameter

### DISCUSSION THREAD

**Discussion by alexander-soare** [pytest-dev/pytest#10762] [Q&A]
I'm aware of `pytest.mark.usefixtures`, and I find it useful for the very reasons stated in the documentation: if your test does not use the return value of the fixture, but depends on some global effects it produces, then it's not sensible to have it as a dummy argument, even if you wish to request it.

How do I solve the same problem if I need my fixture to request another fixture, but it's not sensible to pass the requested fixture is a parameter?

**Comment by portyu9** [comment]
Hi @alexander-soare 

If the dependency is known in advance, passing it as an argument is still the recommended pytest pattern:

```python
@pytest.fixture
def my_fixture(other_fixture):
    # other_fixture does not have to be referenced here
    ...
```

Declaring it in the signature tells pytest that `other_fixture` must be set up before `my_fixture`.

If the fixture really needs to be requested dynamically, use `request.getfixturevalue()`:

```python
@pytest.fixture
def my_fixture(request):
    request.getfixturevalue("other_fixture")
    ...
```

The second form is useful when the fixture name is selected dynamically. For a fixed dependency, I would keep it in the function signature because pytest can see the dependency graph directly.

Please mark this answer as accepted if it helped, thank you.
